worker/rpc/server/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+/*
+    Copyright 2019 Supercomputing Systems AG
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+use std::net::SocketAddr;
+
+use jsonrpsee::{
+    types::error::CallError,
+    ws_server::{RpcModule, WsServerBuilder},
+};
+use log::debug;
+use parity_scale_codec::Encode;
+use tokio::net::ToSocketAddrs;
+
+use substratee_enclave_api::EnclaveApi;
+use substratee_worker_primitives::block::SignedBlock;
+use substratee_worker_primitives::RpcRequest;
+
+#[cfg(test)]
+mod tests;
+#[cfg(test)]
+mod mock;
+
+pub async fn run_server<Enclave>(
+    addr: impl ToSocketAddrs,
+    enclave: Enclave,
+) -> anyhow::Result<SocketAddr>
+where
+    Enclave: EnclaveApi,
+{
```

Comment:
They are semantically the same. Usually you use `where Enclave: EnclaveApi` when there are many trait bounds to satifsfy because the function definition and the trait bound constraints are then more separate to the reader. But here it would be more compact to use `run_server<Enclave: EnclaveApi>` and one could argue that it is better in this case.